client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -41,20 +46,96 @@ export class TeaListContainer extends React.Component {
 
       const revSortOrder = [...newSortOrder].reverse();
 
-      currentState = {
+      currentSortState = {
         columnName: columnName,
         asc: newSortOrder,
         desc: revSortOrder
       };
     }
     this.setState({
       sortedIDs: newSortOrder,
-      memoizedIDs: currentState,
+      memoizedIDs: currentSortState,
       sortColumn: columnName,
       sortOrder: sortOrder
     });
   };
 
+  filterChangeHandler = event => {
+    const name = event.target.name;
+    const value = event.target.value;
+
+    this.setState({
+      ...this.state.formControls,
+      formControls: {
+        ...this.state.formControls,
+        [name]: value
+      }
+    });
+  };
+
+  handleFilterClick = event => {
+    event.preventDefault();
+    const list = this.state.sortedIDs;
+    let currentFilterState = this.state.memoizedIDs;
+    let newFilterOrder;
+
+    if (
+      this.state.formControls.filterCategory !== "" &&
+      this.state.formControls.filterCriteria !== ""
+    ) {
+      newFilterOrder = list.filter(item => {
+        // this returns NaN to get a falsy value if the Number function doesn't return an integer
+        if (Number(this.state.formControls.filterCriteria)) {
+          // I am actively leveraging the coercion of the double equals here
+          return (
+            this.props.teas.allTeas[item][
+              this.state.formControls.filterCategory
+            ] == this.state.formControls.filterCriteria
+          );
+        } else {
+          return (
+            this.props.teas.allTeas[item][
+              this.state.formControls.filterCategory
+            ]
+              .toLowerCase()
+              .search(this.state.formControls.filterCriteria.toLowerCase()) !==
+            -1
+          );
+        }
+      });
+      const revSortOrder = [...newFilterOrder].reverse();
+
+      currentFilterState = {
+        asc: newFilterOrder,
+        desc: revSortOrder
+      };
+
+      this.setState({
+        sortedIDs: newFilterOrder,
+        memoizedIDs: currentFilterState,
+        filtered: true
+      });
+    }
+  };
+
+  handleClearFilterClick = () => {
+    this.setState({
+      sortedIDs: this.props.teas.teaIDs,
+      memoizedIDs: this.props.teas.teaIDs,
+      formControls: {
+        filterCategory: "",
+        filterCriteria: ""
+      },
+      filtered: false
+    });
+  };
+
+  getOptionsFromTeas = (teas, category) => {
```

Comment:
It feels cumbersome to pass props like this down several levels to the component that actually needs it.  In this case, you define `getOptionsFromTeas` here, then pass it to `TeaList` (1) as a prop, then to `TableFilter` (2) and finally to `DataList` (3), where it is finally used.  So the logic is separated by 3 levels from its use case.  

There are a couple ways around this.  Context would work but is not really suitable, since there is only one component that needs this data and not many components that would benefit from a shared context.  Instead, this might be a good place to think about component composition (https://reactjs.org/docs/composition-vs-inheritance.html#containment). You can check out the branch https://github.com/jenniferlynparsons/cuppa-webpack/tree/filter-teas-refactor to see an example of composition in this case (don't feel like you need to copy that code, it's just an example).